### PR TITLE
BXMSPROD-710 skipTests treatment

### PR DESCRIPTION
**Pull request by Ginxo**
BXMSPROD-710
Depends on https://github.com/kiegroup/jenkins-pipeline-shared-libraries/pull/18

### CODE REVIEW

File: downstream.production.stages [hunk 1]
```diff
@@ -14,6 +14,6 @@ stage('Downstream Production Build') {
 
         println "File ${REPOSITORY_LIST_FILE} and ${PRODUCTION_PROJECT_LIST} jenkins file merged in ${projectCollection}"
 
-        treebuild.downstreamBuild(projectCollection, "${SETTINGS_XML_ID}", 'clean install -Dproduction', false)
+        treebuild.downstreamBuild(projectCollection, "${SETTINGS_XML_ID}", 'clean install -Dproductized=true')
```
Comment (mareknovotny) [inline downstream.production.stages:17]: good catch

File: upstream.stages [hunk 2]
```diff
@@ -8,6 +8,6 @@ stage('Upstream Build') {
         def gitURL = env.ghprbAuthorRepoGitUrl ?: env.GIT_URL
         def project = treebuild.getProjectGroupName(treebuild.getProject(gitURL))[1]
 
-        treebuild.upstreamBuild(projectCollection, project, "${SETTINGS_XML_ID}", 'clean install', false)
+        treebuild.upstreamBuild(projectCollection, project, "${SETTINGS_XML_ID}", '-e -nsu -fae -Pbusiness-central,wildfly,sourcemaps,no-showcase clean install -Dproduction -Dfull=true -Dcontainer=wildfly -Dcontainer.profile=wildfly -Dintegration-tests=true -Dmaven.test.failure.ignore=true -Dmaven.test.redirectTestOutputToFile=true -Dgwt.compiler.localWorkers=1')
```
Comment (mareknovotny) [inline upstream.stages:11]: Is this correct? 
I think original upstream maven command was simple just maven clean install for getting the bits built
see https://github.com/kiegroup/kie-jenkins-scripts/blob/master/job-dsls/jobs/pr_jobs.groovy#L12 line and the command from it here:
```
"-B -e -T1C -s \$SETTINGS_XML_FILE -Dkie.maven.settings.custom=\$SETTINGS_XML_FILE -DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true -Denforcer.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true clean install
```
The maven command here is for downstream builds

Comment (Ginxo) [reply upstream.stages:11]: ok, reverting it...

Comment (mareknovotny) [reply upstream.stages]: but are the enforcer and some others properties applied with just short `mvn clean install` it should not be running such plugin execution ;)

Comment (Ginxo) [reply upstream.stages]: @mareknovotny the final mvn command is defined here 
https://github.com/kiegroup/jenkins-pipeline-shared-libraries/blob/f49ed111275fc503ae3cf03d0027efcf098a3426/vars/maven.groovy#L11
so `mvn -B -s $MAVEN_SETTINGS_XML -fae clean install` will be executed. Is it right?

Comment (mareknovotny) [reply upstream.stages]: honestly not sure as I can't see these properties/options which takes executions not wanted in upstream builds
`-DskipTests -Dgwt.compiler.skip=true -Dgwt.skipCompilation=true -Denforcer.skip=true -Dcheckstyle.skip=true -Dspotbugs.skip=true -Drevapi.skip=true`

**Pull request merged**